### ISSUE TITLE
ImportError: No module named googlemaps

### ISSUE THREAD

**Issue by Cephis** [PokemonGoF/PokemonGo-Bot#1040]
### Expected Behavior
Starting pokemon bot

### Actual Behavior
Traceback (most recent call last):
  File "pokecli.py", line 39, in <module>
    from pokemongo_bot import logger
  File "/home/cephis/PokemonGo-Bot-master/pokemongo_bot/__init__.py", line 4, in <module>
    import googlemaps
ImportError: No module named googlemaps

### Steps to Reproduce
I just change personnal values in config.json and release_config.json and start it with python pokecli.py in a terminal

### Other Information
OS: linux mint rosa
Git Commit:

**Comment by DanielHeckrath** [comment]
Have your installed all dependencies with pip before running? Or did your maybe install the in virtualenv and forgot to activate venv before starting the bot with `source bin/activate`?

**Comment by kevinxh** [comment]
It is not a problem of bot, it is your python config problem.

Were this installation step successful: `pip install -r requirements.txt`

**Comment by Cephis** [comment]
This is what this command is telling to me:

`pip install -r requirements.txt
Obtaining pgoapi from git+https://github.com/tejado/pgoapi.git#egg=pgoapi (from -r requirements.txt (line 1))
  Cloning https://github.com/tejado/pgoapi.git to ./src/pgoapi
  Running setup.py (path:/home/cephis/PokemonGo-Bot-master/src/pgoapi/setup.py) egg_info for package pgoapi
    
    warning: no previously-included files matching '*.dll' found anywhere in distribution
    warning: no previously-included files matching '*.pyc' found anywhere in distribution
    warning: no previously-included files matching '*.pyo' found anywhere in distribution
    warning: no previously-included files matching '*.so' found anywhere in distribution
  Installing extra requirements: 'egg'
Downloading/unpacking geopy==1.11.0 (from -r requirements.txt (line 2))
  Downloading geopy-1.11.0-py2.py3-none-any.whl (66kB): 66kB downloaded
Downloading/unpacking protobuf==3.0.0b4 (from -r requirements.txt (line 3))
  Downloading protobuf-3.0.0b4.tar.gz (210kB): 210kB downloaded
  Running setup.py (path:/tmp/pip_build_cephis/protobuf/setup.py) egg_info for package protobuf
    
    no previously-included directories found matching 'google/protobuf/internal/import_test_package'
    warning: no previously-included files found matching 'google/protobuf/internal/*_pb2.py'
    warning: no previously-included files found matching 'google/protobuf/internal/*_test.py'
    warning: no previously-included files found matching 'google/protobuf/internal/*.proto'
    warning: no previously-included files found matching 'google/protobuf/internal/test_util.py'
    warning: no previously-included files matching '*_test.py' found under directory 'google'
    warning: no previously-included files matching '*_test.proto' found under directory 'google'
    warning: no previously-included files matching 'unittest*_pb2.py' found under directory 'google'
    warning: no previously-included files matching '*.dll' found anywhere in distribution
    warning: no previously-included files matching '*.pyc' found anywhere in distribution
    warning: no previously-included files matching '*.pyo' found anywhere in distribution
    warning: no previously-included files matching '*.so' found anywhere in distribution
Downloading/unpacking requests==2.10.0 (from -r requirements.txt (line 4))
  Downloading requests-2.10.0-py2.py3-none-any.whl (506kB): 506kB downloaded
Downloading/unpacking s2sphere==0.2.4 (from -r requirements.txt (line 5))
  Downloading s2sphere-0.2.4-py2.py3-none-any.whl
Downloading/unpacking gpsoauth==0.3.0 (from -r requirements.txt (line 6))
  Downloading gpsoauth-0.3.0.tar.gz
  Running setup.py (path:/tmp/pip_build_cephis/gpsoauth/setup.py) egg_info for package gpsoauth
    
Downloading/unpacking protobuf-to-dict==0.1.0 (from -r requirements.txt (line 7))
  Downloading protobuf-to-dict-0.1.0.tar.gz
  Running setup.py (path:/tmp/pip_build_cephis/protobuf-to-dict/setup.py) egg_info for package protobuf-to-dict
    
    Installed /tmp/pip_build_cephis/protobuf-to-dict/.eggs/nosexcover-1.0.10-py2.7.egg
    Searching for coverage
    Reading https://pypi.python.org/simple/coverage/
    Downloading https://pypi.python.org/packages/ca/76/75cb8a80084dc08d373ef39a04237a658e6dcf4adeaba3fe2a9753a904e3/coverage-4.2b1.tar.gz#md5=587019c5f1a0af28c08e9e9586d5a60a
    Best match: coverage 4.2b1
    Processing coverage-4.2b1.tar.gz
    Writing /tmp/easy_install-sJQylI/coverage-4.2b1/setup.cfg
    Running coverage-4.2b1/setup.py -q bdist_egg --dist-dir /tmp/easy_install-sJQylI/coverage-4.2b1/egg-dist-tmp-RbnNgs
    warning: no previously-included files found matching 'ci/appveyor.token'
    no previously-included directories found matching 'doc/_build'
    no previously-included directories found matching 'tests/eggsrc/build'
    no previously-included directories found matching 'tests/eggsrc/dist'
    no previously-included directories found matching 'tests/eggsrc/*.egg-info'
    warning: no previously-included files matching '*.py[co]' found anywhere in distribution
    In file included from coverage/ctracer/datastack.c:4:0:
    coverage/ctracer/util.h:7:20: fatal error: Python.h: Aucun fichier ou dossier de ce type
     #include <Python.h>
                        ^
    compilation terminated.
    warning: no previously-included files found matching 'ci/appveyor.token'
    no previously-included directories found matching 'doc/_build'
    no previously-included directories found matching 'tests/eggsrc/build'
    no previously-included directories found matching 'tests/eggsrc/dist'
    no previously-included directories found matching 'tests/eggsrc/*.egg-info'
    warning: no previously-included files matching '*.py[co]' found anywhere in distribution
    **
    ** Couldn't install with extension module, trying without it...
    ** BuildFailed: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    **
    Extracting coverage-4.2b1-py2.7.egg to /tmp/pip_build_cephis/protobuf-to-dict/.eggs
    
    Installed /tmp/pip_build_cephis/protobuf-to-dict/.eggs/coverage-4.2b1-py2.7.egg
    Searching for nose>=1.0
    Reading https://pypi.python.org/simple/nose/
    Downloading https://pypi.python.org/packages/58/a5/0dc93c3ec33f4e281849523a5a913fa1eea9a3068acfa754d44d88107a44/nose-1.3.7.tar.gz#md5=4d3ad0ff07b61373d2cefc89c5d0b20b
    Best match: nose 1.3.7
    Processing nose-1.3.7.tar.gz
    Writing /tmp/easy_install-zx12AP/nose-1.3.7/setup.cfg
    Running nose-1.3.7/setup.py -q bdist_egg --dist-dir /tmp/easy_install-zx12AP/nose-1.3.7/egg-dist-tmp-kaS3NH
    no previously-included directories found matching 'doc/.build'
    Extracting nose-1.3.7-py2.7.egg to /tmp/pip_build_cephis/protobuf-to-dict/.eggs
    
    Installed /tmp/pip_build_cephis/protobuf-to-dict/.eggs/nose-1.3.7-py2.7.egg
    
Downloading/unpacking googlemaps==2.4.4 (from -r requirements.txt (line 8))
  Downloading googlemaps-2.4.4-py2-none-any.whl
Downloading/unpacking colorama==0.3.7 (from -r requirements.txt (line 9))
  Downloading colorama-0.3.7-py2.py3-none-any.whl
Downloading/unpacking enum34==1.1.6 (from -r requirements.txt (line 10))
  Downloading enum34-1.1.6-py2-none-any.whl
Downloading/unpacking pyyaml==3.11 (from -r requirements.txt (line 11))
  Downloading PyYAML-3.11.zip (371kB): 371kB downloaded
  Running setup.py (path:/tmp/pip_build_cephis/pyyaml/setup.py) egg_info for package pyyaml
    
Downloading/unpacking haversine==0.4.5 (from -r requirements.txt (line 12))
  Downloading haversine-0.4.5.tar.gz
  Running setup.py (path:/tmp/pip_build_cephis/haversine/setup.py) egg_info for package haversine
    
Downloading/unpacking polyline==1.3.1 (from -r requirements.txt (line 13))
  Downloading polyline-1.3.1-py2.py3-none-any.whl
Requirement already satisfied (use --upgrade to upgrade): six in /usr/lib/python2.7/dist-packages (from pgoapi->-r requirements.txt (line 1))
Requirement already satisfied (use --upgrade to upgrade): setuptools in /usr/local/lib/python2.7/dist-packages (from protobuf==3.0.0b4->-r requirements.txt (line 3))
Downloading/unpacking future (from s2sphere==0.2.4->-r requirements.txt (line 5))
  Downloading future-0.15.2.tar.gz (1.6MB): 1.6MB downloaded
  Running setup.py (path:/tmp/pip_build_cephis/future/setup.py) egg_info for package future
    
    warning: no files found matching '*.au' under directory 'tests'
    warning: no files found matching '*.gif' under directory 'tests'
    warning: no files found matching '*.txt' under directory 'tests'
Downloading/unpacking pycryptodomex>=3.0 (from gpsoauth==0.3.0->-r requirements.txt (line 6))
  Downloading pycryptodomex-3.4.2.tar.gz (6.5MB): 6.5MB downloaded
  Running setup.py (path:/tmp/pip_build_cephis/pycryptodomex/setup.py) egg_info for package pycryptodomex
    
Installing collected packages: pgoapi, geopy, protobuf, requests, s2sphere, gpsoauth, protobuf-to-dict, googlemaps, colorama, enum34, pyyaml, haversine, polyline, future, pycryptodomex
  Running setup.py develop for pgoapi
    error: can't create or remove files in install directory
    
    The following error occurred while trying to add or remove files in the
    installation directory:
    
        [Errno 13] Permission denied: '/usr/local/lib/python2.7/dist-packages/test-easy-install-11602.write-test'
    
    The installation directory you specified (via --install-dir, --prefix, or
    the distutils default setting) was:
    
        /usr/local/lib/python2.7/dist-packages/
    
    Perhaps your account does not have write access to this directory?  If the
    installation directory is a system-owned directory, you may need to sign in
    as the administrator or "root" account.  If you do not have administrative
    access to this machine, you may wish to choose a different installation
    directory, preferably one that is listed in your PYTHONPATH environment
    variable.
    
    For information on other options, you may wish to consult the
    documentation at:
    
      https://pythonhosted.org/setuptools/easy_install.html
    
    Please make the appropriate changes for your system and try again.
    
    Complete output from command /usr/bin/python -c "import setuptools, tokenize; __file__='/home/cephis/PokemonGo-Bot-master/src/pgoapi/setup.py'; exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" develop --no-deps:
    running develop

error: can't create or remove files in install directory



The following error occurred while trying to add or remove files in the

installation directory:



    [Errno 13] Permission denied: '/usr/local/lib/python2.7/dist-packages/test-easy-install-11602.write-test'



The installation directory you specified (via --install-dir, --prefix, or

the distutils default setting) was:



    /usr/local/lib/python2.7/dist-packages/



Perhaps your account does not have write access to this directory?  If the

installation directory is a system-owned directory, you may need to sign in

as the administrator or "root" account.  If you do not have administrative

access to this machine, you may wish to choose a different installation

directory, preferably one that is listed in your PYTHONPATH environment

variable.



For information on other options, you may wish to consult the

documentation at:



  https://pythonhosted.org/setuptools/easy_install.html



Please make the appropriate changes for your system and try again.



----------------------------------------
Cleaning up...
Command /usr/bin/python -c "import setuptools, tokenize; __file__='/home/cephis/PokemonGo-Bot-master/src/pgoapi/setup.py'; exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" develop --no-deps failed with error code 1 in /home/cephis/PokemonGo-Bot-master/src/pgoapi
Storing debug log for failure in /home/cephis/.pip/pip.log
`

**Comment by kevinxh** [comment]
@Cephis `sudo pip install -r requirements.txt`

**Comment by Cephis** [comment]
cephis@Raven ~ $ sudo pip install -r requirements.txt
[sudo] password for cephis: 
Could not open requirements file: [Errno 2] No such file or directory: 'requirements.txt'
Storing debug log for failure in /home/cephis/.pip/pip.log

**Comment by MilkyPL** [comment]
I'm getting the same error. Installation of requirements.txt was succesfull:
`jakub@jakub-arch ~/PokemonGo-Bot> sudo pip install -r requirements.txt
Obtaining pgoapi from git+https://github.com/tejado/pgoapi.git#egg=pgoapi (from -r requirements.txt (line 1))
  Updating ./src/pgoapi clone
Requirement already satisfied (use --upgrade to upgrade): geopy==1.11.0 in /usr/lib/python3.5/site-packages (from -r requirements.txt (line 2))
Requirement already satisfied (use --upgrade to upgrade): protobuf==3.0.0b4 in /usr/lib/python3.5/site-packages (from -r requirements.txt (line 3))
Requirement already satisfied (use --upgrade to upgrade): requests==2.10.0 in /usr/lib/python3.5/site-packages (from -r requirements.txt (line 4))
Requirement already satisfied (use --upgrade to upgrade): s2sphere==0.2.4 in /usr/lib/python3.5/site-packages (from -r requirements.txt (line 5))
Requirement already satisfied (use --upgrade to upgrade): gpsoauth==0.3.0 in /usr/lib/python3.5/site-packages (from -r requirements.txt (line 6))
Requirement already satisfied (use --upgrade to upgrade): protobuf-to-dict==0.1.0 in /usr/lib/python3.5/site-packages (from -r requirements.txt (line 7))
Requirement already satisfied (use --upgrade to upgrade): googlemaps==2.4.4 in /usr/lib/python3.5/site-packages (from -r requirements.txt (line 8))
Requirement already satisfied (use --upgrade to upgrade): colorama==0.3.7 in /usr/lib/python3.5/site-packages (from -r requirements.txt (line 9))
Requirement already satisfied (use --upgrade to upgrade): enum34==1.1.6 in /usr/lib/python3.5/site-packages (from -r requirements.txt (line 10))
Requirement already satisfied (use --upgrade to upgrade): pyyaml==3.11 in /usr/lib/python3.5/site-packages (from -r requirements.txt (line 11))
Requirement already satisfied (use --upgrade to upgrade): haversine==0.4.5 in /usr/lib/python3.5/site-packages (from -r requirements.txt (line 12))
Requirement already satisfied (use --upgrade to upgrade): polyline==1.3.1 in /usr/lib/python3.5/site-packages (from -r requirements.txt (line 13))
Requirement already satisfied (use --upgrade to upgrade): six in /usr/lib/python3.5/site-packages (from pgoapi->-r requirements.txt (line 1))
Requirement already satisfied (use --upgrade to upgrade): setuptools in /usr/lib/python3.5/site-packages (from protobuf==3.0.0b4->-r requirements.txt (line 3))
Requirement already satisfied (use --upgrade to upgrade): future in /usr/lib/python3.5/site-packages (from s2sphere==0.2.4->-r requirements.txt (line 5))
Requirement already satisfied (use --upgrade to upgrade): pycryptodomex>=3.0 in /usr/lib/python3.5/site-packages (from gpsoauth==0.3.0->-r requirements.txt (line 6))
Installing collected packages: pgoapi
  Running setup.py develop for pgoapi
Successfully installed pgoapi-1.1.0`
The weird thing is that every time I execute
`sudo pip install -r requirements.txt`
It prints out
`Installing collected packages: pgoapi
  Running setup.py develop for pgoapi
Successfully installed pgoapi-1.1.0`
I'm using Arch Linux

**Comment by MilkyPL** [comment]
I managed to fix it by executing `sudo pip2.7 install -r requirements.txt`
It turns out that you may have installed wrong version of pip (python3).

**Comment by Cephis** [comment]
It looks like I don't have the wrong version, tell me if you think of something else, I follow this project...

cephis@Raven ~ $ pip --version
pip 1.5.4 from /usr/lib/python2.7/dist-packages (python 2.7)

**Comment by mrwbsn** [comment]
use  `virtualenv -p <YOUR PYTHON 2.7 PATH> <DEST PATH>` when creating virtualenv to make sure you use python2.7, i use `virtualenv -p /user/bin/python2.7 .`

**Comment by TheGoldenXY** [comment]
Please upgrade to the latest `dev` branch by running this command:
```
git reset --hard
git pull -f
```

Then run this command:
`pip install --upgrade -r requirements.txt`

If issue persists, open a new issue.

/closed
